### DISCUSSION TITLE
ESP32-S3 with RM68120 ... black screen !

### DISCUSSION THREAD

**Discussion by phil31** [Bodmer/TFT_eSPI#2092] [Compatible displays and setup files]
Hello, 
i'm trying the TFT_eSPI library with plateformIO, on the board ESP32-S3-HMI ( https://github.com/W00ng/ESP32-S3-HMI-DevKit ).
ESP32-S3 with RM68120 driver (lib_deps = bodmer/TFT_eSPI@^2.4.78).
 i double check pinout of the LCD to GPIO !
the LCD is mapped with 16 bits data, i try it on both mode 8 and 16 bits.
the board was delivery with a working example from LVGL .. 

![image](https://user-images.githubusercontent.com/9626218/196050980-2066cd28-e63c-4d91-bd7d-4015e2a79a10.png)

totaly newbie with this library, maybe i forget something .. 
i just create a new "Setup999_RM68120_parallel.h" file  and define it in the "User_Setup_Select.h" file : 
#include <User_Setups/Setup999_RM68120_Parallel.h>

--------------------------------------------------------------------
`// See SetupX_Template.h for all options available
#define USER_SETUP_ID 999

#define ESP32_PARALLEL
//#define TFT_PARALLEL_8_BIT
// #define TFT_PARALLEL_16_BIT
#define RM68120_DRIVER

// The ESP32 and TFT the pins used for testing are:
#define TFT_CS   -1  // Chip select control pin (library pulls permanently low
#define TFT_DC   38  // Data Command control pin - must use a pin in the range 0-31
#define TFT_RST  21  // Reset pin, toggles on startup

#define TFT_WR    17  // Write strobe control pin - must use a pin in the range 0-31
#define TFT_RD    -1  // Read strobe control pin - RD pin permanently connected to 3V3

#define TFT_D0   1  // Must use pins in the range 0-31 for the data bus
#define TFT_D1   9  // so a single register write sets/clears all bits.
#define TFT_D2   2  // Pins can be randomly assigned, this does not affect
#define TFT_D3   10  // TFT screen update performance.
#define TFT_D4   3
#define TFT_D5   11
#define TFT_D6   4
#define TFT_D7   12
// #define TFT_D8   5  // Must use pins in the range 0-31 for the data bus
// #define TFT_D9   13  // so a single register write sets/clears all bits.
// #define TFT_D10  6  // Pins can be randomly assigned, this does not affect
// #define TFT_D11  14  // TFT screen update performance.
// #define TFT_D12  7
// #define TFT_D13  15
// #define TFT_D14  8
// #define TFT_D15  16

#define LOAD_GLCD   // Font 1. Original Adafruit 8 pixel font needs ~1820 bytes in FLASH
#define LOAD_FONT2  // Font 2. Small 16 pixel high font, needs ~3534 bytes in FLASH, 96 characters
#define LOAD_FONT4  // Font 4. Medium 26 pixel high font, needs ~5848 bytes in FLASH, 96 characters
#define LOAD_FONT6  // Font 6. Large 48 pixel font, needs ~2666 bytes in FLASH, only characters 1234567890:-.apm
#define LOAD_FONT7  // Font 7. 7 segment 48 pixel font, needs ~2438 bytes in FLASH, only characters 1234567890:.
#define LOAD_FONT8  // Font 8. Large 75 pixel font needs ~3256 bytes in FLASH, only characters 1234567890:-.
#define LOAD_GFXFF  // FreeFonts. Include access to the 48 Adafruit_GFX free fonts FF1 to FF48 and custom fonts

#define SMOOTH_FONT
`
--------------------------------------------------------------------

compile success and upload OK, but the screen stay black :(

small main.cpp as : 

`void setup() {
    tft.init();
    tft.fillScreen(TFT_RED);
}

void loop() {
  // put your main code here, to run repeatedly:
}`

Please anybody can confirm that the RM68120 driver is working ? as i read it is "untested" driver ...?

thanks for any tips, regards
Phil

**Comment by Bodmer** [comment]
The display is 16 bit parallel. The library only supports 8 bit parallel on the ESP32 family of processors so  will not work with that board.

**Comment by phil31** [comment]
thanks for your reply, he can work on 8 bits mode too ? 
as the RM68120  driver is provided on the library .. nope ?
from the datasheet : "The RM68120 incorporates command method 24-/16-/8-bits bus display command interface."

**Reply by Bodmer** [reply]
Yes, the IM(3..0) pins on the RM68120 must be set to 0000 to use 8 bit mode. These pins may have been hardwired on the module so changing to 8 bit mode may need surgery to the board and is thus is impractical to modify.

**Comment by phil31** [comment]
ohh Fu... just forget this small detail ! :(

well.. the ESP32-S3 work in 16bits mode ( as this board proov it, using the ESP-IDF framework )
may i ask why you don't write the 16 bits version of the driver ?
not enough time or is there an issue with the library herself ?

thanks for your support and library job !
regards
Phil

**Reply by Bodmer** [reply]
The library was written for the original ESP32 which has far fewer GPIO. 16 bit TFT board modules were also rare.

Adaption for the new ESP32 S2/S3 is much more recent, so.e users have adapted the library to 16 bits but I have been focussing on paid work!

**Comment by phil31** [comment]
of course i understand that you have a life and all stuff  around !

16bits TFT are not "rare", this is a requirement for bigger LCD (in my case, 800x480).
small screen can work easly with SPI bus

do you plan to implement some 16 bits mode for the ESP32 S3 more or less soon ?

thanks
Phil

**Comment by phil31** [comment]
FIY, my board (RM68120 driver) is working with the LovyanGFX library :o)

regards

**Comment by Bodmer** [comment]
Great, glad you have it working. Good luck with your project.
